tools_and_services/setting_up_windows_for_rails.md

```diff
@@ -0,0 +1,170 @@
+# Setting up your Windows 11 machine for running rails projects
+
+## Prerequisites
+
+- Windows 11
+- VS Code
+
+## 1. Install Ubuntu in wsl
+
+Open power shell and run the following command:
+
+```powershell
+wsl --install -d Ubuntu
+```
+
+The command will eventually will tell you to restart your machine. Do that.
+
+After the restart the installation will continue, and eventually it will prompt you for a username and password for your new linux user.\
+With the credentials in place, you will be launched into a linux terminal.
+
+## 2. Install ZSH and Oh-my-zsh
+
+```sh
+sudo apt -y install zsh
+chsh -s $(which zsh)
+```
+
+Open a new terminal window:
+
+1. Press the "⋁" button in the top of the terminal window. Then select "Ubuntu".
+2. Press "0" to setup zsh with an empty `.zshrc`-file
+3. Run `sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"`
+
+Done
+
+## 3. Install asdf for zsh
+
+Install required packages for asdf and ruby-build (which is used by the asdf ruby plugin):
+
+```sh
+sudo apt update # Ensure that you have the latest package metadata
+sudo apt -y install curl git curl autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm6 libgdbm-dev libdb-dev libpq-dev
+```
+
+Then install asdf:
+
+```sh
+git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.14.0
```

Comment:
🤔 maybe a prior step to check what the latest tag is?
```suggestion
You can check the latest version on Github and replace the version number.
```

Reply:
I was myself not entirely happy about this, but it was what I could grab from the asdf docks.